pkg/scheduler/actions/elect/elect.go

```diff
@@ -18,23 +18,23 @@ func New() *Action {
 }
 
 // Name returns the action name
-func (alloc *Action) Name() string {
+func (e *Action) Name() string {
 	return "elect"
 }
 
 // Initialize inits the action
-func (alloc *Action) Initialize() {}
+func (e *Action) Initialize() {}
 
 // Execute selects the target job which is of the highest priority and waits for the longest time.
-func (alloc *Action) Execute(ssn *framework.Session) {
+func (e *Action) Execute(ssn *framework.Session) {
 	klog.V(3).Infof("Enter Elect ...")
 	defer klog.V(3).Infof("Leaving Elect ...")
 
 	if util.Reservation.TargetJob == nil {
 		klog.V(4).Infof("Start select Target Job")
 		var pendingJobs []*api.JobInfo
 		for _, job := range ssn.Jobs {
-			if job.PodGroup.Status.Phase == scheduling.PodGroupPending {
+			if job.PodGroup.Status.Phase == scheduling.PodGroupInqueue {
```

Comment:
The job, whose podgroup is `inqueue`, is elected as target job. That may be not so reasonable for it will be scheduled at normal process originally.

Reply:
I agree. But some plugins permit inqueue if queue's capacity not set, e.g. proportion. I prefer putting `elect` behind `allocate` and check podgroup in `inqueue` status. At that stage, podgroups in `inqueue` status are starving.

Reply:
I think `overcommit` plugin maybe block some large job change to `inqueue`.  Please try to config `overcommit` plugin.

Reply:
I think we can add another condition : Whether the job has been into the scheduler session but the job's status is not running